### ISSUE TITLE
Import modules from URLs

### ISSUE THREAD

**Issue by noamross** [rticulate/import#8]
It occurred to me from our twitter discussion that one could get some compact "single script reproducibility" by allowing `import::from()` and `import::here()` to take URLs or gist URLs as arguments so that your modules can be placed anywhere on the web, e.g., 

```
import::from('https://gist.github.com/noamross/378884a472b5ef58c6cf", myfunc)
```

**Comment by smbache** [comment]
This has been in my mind some; I think it is a good idea, but I'm still wondering about a few things.

For files, `import` is able to infer whether reloading is necessary (if a file changed); this is not really possible for URLs, but makes it more efficient. Also, should files be downloaded, or only read? One could also download to a temporary location... Or to working directory, and reloads will happen if they are deleted...

Ideas are welcome.

**Comment by hyiltiz** [comment]
I think downloading to the temporary directory is a good idea. There is no way to know if the file has changed at the URL; in fact, it may be preferable to ask for confirmation when downloading a different file before overwriting the old one. I am not sure how we can "only read" the file without downloading first. 

Lastly, there are implications for security. An unsuspecting user might `import` a unsuspicious program from a URL that may even import its own malicious program from another URL. I wonder how `golang`'s import mechanics deal with this potential security issue.

**Comment by torfason** [comment]
This seems like a good use case for the `pins` package, which is intended to solve exactly this issue (about whether remote files have changed, and downloading them into temporary files that are used if the remote is unchanged or not accessible). As in:

```
> mypackPin = pins::pin("https://gist.github.com/noamross/378884a472b5ef58c6cf")
> import::from(mypackPin, myfunc)
```

That said, the above code fails for me with `Error: there is no package called ‘mypackPin’`, but that seems a smaller issue than implementing downloads, and is probably (I'm guessing) related with some non-standard evaluation issue. In fact , this is just the same error I get if I assign the name of the file to a string before passing it to `import::from()` instead of hard-coding it:

```
> mypackString = "mypack.R"
> import::from(mypackString, myfunc)
Error: there is no package called ‘mypackString’
```

**Comment by torfason** [comment]
After some further digging, it seems that it is indeed possible to resolve this using the `pins` package. A reproducible example of how the original use case listed by @noamross can be implemented follows:

``` r
url <- "https://gist.githubusercontent.com/noamross/378884a472b5ef58c6cf/raw/1cd3a47158427ef7e2faa897d32821f14fa19bfd/test.R"
import::from(pins::pin(url), "myfunc", .character_only=TRUE)
myfunc(3)
#> [1] 4
```

<sup>Created on 2020-06-25 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by torfason** [comment]
With the latest fixes to using absolute file names for modules, this is considered fixed in master, to be closed on CRAN release. Comments and tests welcome.

**Comment by torfason** [comment]
Fixed in release 1.2.0
